init/Kconfig

```
@@ -58,7 +58,7 @@ config LLD_VERSION
 	default 0
 
 config HAS_RUST
-	depends on ARM64 || (PPC64 && CPU_LITTLE_ENDIAN) || X86_64
+	depends on ARM64 || (PPC64 && CPU_LITTLE_ENDIAN) || X86_64 || ARM
```

Just a nit: please put it first to keep it the archs alphabetically sorted (same in a few other places).

```suggestion
	depends on ARM || ARM64 || (PPC64 && CPU_LITTLE_ENDIAN) || X86_64
```

---

The most vanilla 32-bit arm triple `arm-unknown-linux-gnueabi` appears to make Rust generate code compatible with `armv6` and up, in non-Thumb mode. I will try to make sure that `HAS_RUST` appears only when the kernel is built with a compatible instruction set.

---

Or, force Rust to generate lowest-common-denominator arm instructions, so it'll run on each and every arm core out there. But that will result in very bad performance on newer cores.

@ojeda we seem to have the following trade-off:
a) use newer instruction set for good performance on newer cores, but won't work on older cores (where `HAS_RUST` will be automatically disabled)
b) use lowest-common-denominator instruction set which will run on all cores, but results in bad performance on older cores

What is your opinion?

---

Do not worry too much about that -- we will have the ability to mod the target soon enough. So it is fine picking either one.

I think the lowest-common-denominator makes the most sense since anyhow we are not measuring performance yet.

---

Supporting the newer instruction set turned out to be the path of least resistance...